### PR TITLE
Draft PR to get Feedback on component created. 

### PR DESCRIPTION
**Pull request by AshishMotanamGurunadham**
### Summary

Draft PR to get feedback on component created for the line graph. 

### Deployment Link

https://terra-graphs-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

**Comment by yuderekyu** [comment]
you can convert this to a draft pr with the "convert to draft pr" under the reviewers section: https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/

File: packages/terra-graphs/src/components/Line/LineGraph.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import Carbon from '../../../../carbon-graphs/src/js/carbon';
+import '../../css/Graph.module.scss';
+import '../../css/LineGraph.module.scss';
+import utils from '../../../../carbon-graphs/src/js/helpers/utils';
+
+class LineGraph extends React.Component {
```
Comment (tbiethman) [inline packages/terra-graphs/src/components/Line/LineGraph.jsx:7]: At this point I'd recommend writing functional components with hooks: https://reactjs.org/docs/hooks-effect.html

```jsx
const LineGraph = ({ graphID, graphConfig, dataset }) => {
  React.useEffect(() => {
      const graph = Carbon.api.graph(this.props.graphConfig);
      (utils.isArray(this.props.dataset)) ? (this.props.dataset.forEach((data) => {
        graph.loadContent(Carbon.api.line(data));
      })) : graph.loadContent(Carbon.api.line(this.props.dataset));
  }, [graphConfig, dataset]);

  return (
    <div id="canvasContainer">
      <div id="tooltip" className="tooltip" style={{ display: 'none' }} />
      <div id={this.props.graphID} />
    </div>
  );
};


```

File: packages/terra-graphs/src/components/Line/LineGraph.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  render() {
+    debugger;
+    return (
+      <div id="canvasContainer">
+        <div id="tooltip" className="tooltip" style={{ display: 'none' }} />
```
Comment (tbiethman) [inline packages/terra-graphs/src/components/Line/LineGraph.jsx:12]: You likely want to make this id unique on a per-component basis. You can do that by combining 'tooltip' with the provided graphID, or by generating an ID (we typically use the uuid package to do that).

File: packages/terra-graphs/src/components/Line/LineGraph.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <div id={this.props.graphID} />
+      </div>
+    );
+  }
+
+  componentDidMount() {
+    debugger;
+    const graph = Carbon.api.graph(this.props.graphConfig);
+    (utils.isArray(this.props.dataset)) ? (this.props.dataset.forEach((data) => {
```
Comment (tbiethman) [inline packages/terra-graphs/src/components/Line/LineGraph.jsx:21]: I'd go with an if/else here vs. an inline ternary but I'm not sure if the linter has an opinion or not.

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 1 above)
Comment (yuderekyu) [inline packages/terra-graphs/src/components/Line/LineGraph.jsx:7]: we can type check props using `prop-types`. here's an example in one of our components: https://github.com/cerner/terra-application/blob/main/src/application-base/ApplicationBase.jsx#L33

https://reactjs.org/docs/typechecking-with-proptypes.html

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 2 above)
Comment (yuderekyu) [inline packages/terra-graphs/src/components/Line/LineGraph.jsx:12]: is this supposed to default to being hidden? we're using sass for component styles, so best to move inline styles there, then apply a specific css class via className

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 3 above)
Comment (yuderekyu) [inline packages/terra-graphs/src/components/Line/LineGraph.jsx:21]: should we tighten the api and expect the dataset to always be an array? 

is this logic standard for all graphs? if so, we could convert it into a custom useEffect hook, then import it into components: https://reactjs.org/docs/hooks-custom.html

terra example: https://github.com/cerner/terra-application/blob/d6687c6d0ac87684150063a22ae0e3936e09f92b/src/application-base/private/useTestOverrides.js

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 1 above)
Comment (AshishMotanamGurunadham) [reply packages/terra-graphs/src/components/Line/LineGraph.jsx]: Fixed [here](https://github.com/cerner/terra-graphs/pull/74/commits/631c1818a81503e72ba2f63e2f41eb9df667ba04#diff-5e17603406b642035820d408335769ae48391021a7787fd86404b130590e9606R14-R34).

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 3 above)
Comment (AshishMotanamGurunadham) [reply packages/terra-graphs/src/components/Line/LineGraph.jsx]: Fixed [here](https://github.com/cerner/terra-graphs/pull/74/commits/631c1818a81503e72ba2f63e2f41eb9df667ba04#diff-5e17603406b642035820d408335769ae48391021a7787fd86404b130590e9606R17-R23).

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 1 above)
Comment (AshishMotanamGurunadham) [reply packages/terra-graphs/src/components/Line/LineGraph.jsx]: Added propTypes [here](https://github.com/cerner/terra-graphs/pull/74/commits/631c1818a81503e72ba2f63e2f41eb9df667ba04#diff-5e17603406b642035820d408335769ae48391021a7787fd86404b130590e9606R8-R12).

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 2 above)
Comment (AshishMotanamGurunadham) [reply packages/terra-graphs/src/components/Line/LineGraph.jsx]: Fixed [here](https://github.com/cerner/terra-graphs/pull/74/commits/631c1818a81503e72ba2f63e2f41eb9df667ba04#diff-3df99a036edf4bc1ff6a4da7270b378e91ed6043388577b14c68cf9f75d52f68R557-R559).

File: packages/terra-graphs/src/components/Line/LineGraph.jsx (hunk 3 above)
Comment (AshishMotanamGurunadham) [reply packages/terra-graphs/src/components/Line/LineGraph.jsx]: Added useEffect [here](https://github.com/cerner/terra-graphs/pull/74/commits/631c1818a81503e72ba2f63e2f41eb9df667ba04#diff-5e17603406b642035820d408335769ae48391021a7787fd86404b130590e9606R15).

**Pull request closed without merging**